src/Common/src/CoreLib/System/Marvin.cs

```diff
@@ -36,10 +38,10 @@ public static int ComputeHash32(ref byte data, int count, ulong seed)
 
             while (ucount >= 8)
             {
-                p0 += Unsafe.ReadUnaligned<uint>(ref Unsafe.AddByteOffset(ref data, byteOffset));
+                p0 += Unsafe.ReadUnaligned<uint>(ref Unsafe.AddByteOffset(ref data, (IntPtr)byteOffset));
```

Comment:
These IntPtr cast will have overflow checks. I expect that they are introducing pretty signficant performance regression.